apps/service_providers/tracing/service.py

```diff
@@ -72,11 +80,20 @@ def trace(
         user_id: str,
         inputs: dict[str, Any] | None = None,
         metadata: dict[str, str] | None = None,
+        input_message_id: str | None = None,
+        experiment_id: int | None = None,
+        session_id_fk: int | None = None,
+        participant_id: int | None = None,
     ):
         self.trace_id = uuid.uuid4()
         self.trace_name = trace_name
         self.session_id = session_id
         self.user_id = user_id
+        self.experiment_id = experiment_id
+        self._start_time = timezone.now()
```

Comment:
For timing you should use `time.time()` since we don't actually care about dates and timezones etc.

```python
start = time.time()
end = time.time()
duration_ms = int((end - start) * 1000)
```